components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdPManagementServiceComponent.java

```diff
@@ -85,13 +95,50 @@
     private static RealmService realmService = null;
 
     private static ConfigurationContextService configurationContextService = null;
+    private static RegistryService registryService;
```

Comment:
I would prefer to have a dataHolder class to hold instances of RealmService, ConfigurationContextService and RegistryService or have them as private instance variables instead of writing to a static variable from an instance method.